test/sqllogictest/privilege_checks.slt

```diff
@@ -1713,6 +1713,144 @@ DROP TYPE ty;
 ----
 COMPLETE 0
 
+## Explain schema
+
+simple conn=mz_system,user=mz_system
+CREATE TYPE ty AS (a text);
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+REVOKE USAGE ON TYPE ty FROM PUBLIC;
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+CREATE TABLE t (a ty);
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+CREATE CONNECTION kafka_conn TO KAFKA (BROKER 'localhost:9092') WITH (VALIDATE = false);
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+CREATE CONNECTION IF NOT EXISTS csr_conn TO CONFLUENT SCHEMA REGISTRY (URL 'https://google.com');
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+ALTER SYSTEM SET enable_sink_doc_on_option = true;
+----
+COMPLETE 0
+
+simple conn=joe,user=joe
+EXPLAIN VALUE SCHEMA FOR CREATE SINK sink FROM t INTO KAFKA CONNECTION kafka_conn (TOPIC 'topic') KEY (a) NOT ENFORCED FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_conn (DOC ON COLUMN ty.a = 'comment') ENVELOPE UPSERT;
+----
+db error: ERROR: permission denied for SCHEMA "materialize.public"
+
+simple conn=mz_system,user=mz_system
+GRANT USAGE ON SCHEMA materialize.public TO joe;
+----
+COMPLETE 0
+
+simple multiline,conn=joe,user=joe
+EXPLAIN VALUE SCHEMA FOR CREATE SINK sink FROM t INTO KAFKA CONNECTION kafka_conn (TOPIC 'topic') KEY (a) NOT ENFORCED FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_conn (DOC ON COLUMN ty.a = 'comment') ENVELOPE UPSERT;
+----
+{
+  "type": "record",
+  "name": "envelope",
+  "fields": [
+    {
+      "name": "a",
+      "type": [
+        "null",
+        {
+          "type": "record",
+          "name": "record0",
+          "namespace": "com.materialize.sink",
+          "fields": [
+            {
+              "name": "a",
+              "type": [
+                "null",
+                "string"
+              ],
+              "doc": "comment"
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+EOF
+COMPLETE 1
+
+simple multiline,conn=mz_support,user=mz_support
```

Comment:
Added a test for `mz_support`

Reply:
Thanks! There is a file called `mz-support-privileges.slt`, I think the test should go there.